numpy/core/src/multiarray/descriptor.c

```diff
@@ -1546,7 +1546,8 @@ PyArray_DescrConverter(PyObject *obj, PyArray_Descr **at)
 
         /* A typecode like 'd' */
         if (len == 1) {
-            check_num = type[0];
+            /* Python byte string characters are unsigned */
+            check_num = ((unsigned char *)type)[0];
```

Comment:
Not convinced by this, I think it's technically better to do this, but it's an extra cast just to double-cover a case that's already covered, and you'll have bigger problems to fix before you get to >127 built-in dtypes

I left it in to see what you thought

Reply:
I think this would be clearer as `(unsigned char) type[0]`

Reply:
Yes, apologies, casting away signedness in values scares me, and I was too lazy to check the rules, to confirm the equivalent behaviour. I'll update